web/packages/teleport/src/services/websession/websession.ts

```diff
@@ -35,14 +37,31 @@ const logger = Logger.create('services/session');
 let sesstionCheckerTimerId = null;
 
 const session = {
-  logout(rememberLocation = false) {
+  /** logout logs the user out of Teleport, and if SAML SLO is enabled, also log them out of their IdP. */
+  logout(rememberLocation = false, userCtx: StoreUserContext) {
     api.delete(cfg.api.webSessionPath).finally(() => {
-      history.goToLogin(rememberLocation);
+      // If SAML SLO (single logout) is enabled, the user will be redirected to the IdP's SLO URL, after which they will automatically be redirected back
+      // to the login page.
+      if (userCtx?.getIsSamlSloEnabled()) {
+        const sloUrl = userCtx.getSamlSloUrl();
+        window.open(sloUrl, '_self');
+      } else {
+        history.goToLogin(rememberLocation);
+      }
     });
 
     this.clear();
   },
 
+  /** logoutWithoutSlo logs the user out of Teleport, but not their SAML IdP. */
+  logoutWithoutSlo(rememberLocation = false) {
```

Comment:
`logout` (with SLO) should be called whenever we're actually performing the action of logging the user out of a valid session (either because they clicked "Logout" somewhere, or if they are logged out due to inactivity). `logoutWithoutSlo` should only be called when the session is already invalid, for example in `websession.ts`:

```
...
// check if logout was triggered from other tabs
if (storageService.getBearerToken() === null) {
  session.logoutWithoutSlo();
}
...
```

we wouldn't want to trigger the SLO flow in each tab, which would cause every tab to be redirected to the IdP's SLO URL. 

Similar case in `Authenticated.tsx`:

```
    const checkIfUserIsAuthenticated = async () => {
      if (!session.isValid()) {
        logger.warn('invalid session');
        session.logoutWithoutSlo(true /* rememberLocation */);
        return;
      }

      try {
        const result = await session.validateCookieAndSession();
        if (result.hasDeviceExtensions) {
          session.setIsDeviceTrusted();
        }
        setAttempt({ status: 'success' });
      } catch (e) {
        if (e instanceof ApiError && e.response?.status == 403) {
          logger.warn('invalid session');
          session.logoutWithoutSlo(true /* rememberLocation */);
          // No need to update attempt, as `logout` will
          // redirect user to login page.
          return;
        }
    };
```

There is an edge case here where if the Teleport session is expired/invalid then the user won't be logged out of the IdP alongside Teleport. I don't think there's a way around this because to be able to log the user out of the IdP, we need information about the user's SAML session which is only received on SAML login and kept track of within the user's active session.

Reply:
> I think the confusion comes from the fact that even the session is invalid, we still call a log out function that makes a network request.

This was the initial implementation, I've changed this now to prevent the unnecessary extra calls.